compiler/rustc_typeck/src/collect.rs

```diff
@@ -2647,6 +2647,91 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
+                    [NestedMetaItem::MetaItem(set)] => {
+                        if set
+                            .path
+                            .segments
+                            .iter()
+                            .map(|x| x.ident.name)
+                            .eq([sym::arm, sym::a32].iter().cloned())
+                        {
+                            if tcx.sess.target.target.options.has_thumb_interworking {
+                                Some(InstructionSetAttr::ArmA32)
+                            } else {
+                                struct_span_err!(
+                                    tcx.sess.diagnostic(),
+                                    attr.span,
+                                    E0779,
+                                    "target does not support `#[instruction_set]`"
+                                )
+                                .emit();
+                                None
+                            }
+                        } else if set.path.segments.iter().map(|x| x.ident.name).eq([
+                            sym::arm,
+                            sym::t32,
+                        ]
+                        .iter()
+                        .cloned())
+                        {
+                            if tcx.sess.target.target.options.has_thumb_interworking {
+                                Some(InstructionSetAttr::ArmT32)
+                            } else {
+                                struct_span_err!(
+                                    tcx.sess.diagnostic(),
+                                    attr.span,
+                                    E0779,
+                                    "target does not support `#[instruction_set]`"
+                                )
+                                .emit();
+                                None
+                            }
+                        } else {
+                            struct_span_err!(
+                                tcx.sess.diagnostic(),
+                                attr.span,
+                                E0779,
+                                "Invalid instruction set specified",
+                            )
+                            .emit();
+                            None
+                        }
+                    }
+                    [] => {
+                        struct_span_err!(
+                            tcx.sess.diagnostic(),
+                            attr.span,
+                            E0778,
+                            "`#[instruction_set]` requires an argument"
+                        )
+                        .emit();
+                        None
+                    }
+                    _ => {
+                        struct_span_err!(
+                            tcx.sess.diagnostic(),
+                            attr.span,
+                            E0779,
+                            "Cannot specify more than one instruction set"
```

Comment:
Error messages are lower-case

```suggestion
                            "cannot specify more than one instruction set"
```